src/pages/sign-transaction/components/contract-call-details/function-arguments-list.tsx

```diff
@@ -0,0 +1,37 @@
+import React, { memo, Suspense } from 'react';
+import { Stack, StackProps } from '@stacks/ui';
+
+import { Divider } from '@components/divider';
+import { Caption } from '@components/typography';
+import { useTransactionRequestState } from '@store/transactions/requests.hooks';
+
+import { FunctionArgumentItem } from './function-argument-item';
+
+function FunctionArgumentsList(props: StackProps): JSX.Element | null {
+  const transactionRequest = useTransactionRequestState();
+
+  if (!transactionRequest || transactionRequest.txType !== 'contract_call') {
+    return null;
+  }
+  const hasArgs = transactionRequest.functionArgs.length > 0;
+
+  return (
+    <>
+      {hasArgs ? (
+        <Stack divider={<Divider />} spacing="base" {...props}>
+          {transactionRequest.functionArgs.map((arg, index) => {
+            return (
+              <Suspense fallback={<>loading</>} key={`${arg}-${index}`}>
+                <FunctionArgumentItem arg={arg} index={index} />
+              </Suspense>
+            );
+          })}
+        </Stack>
+      ) : (
+        <Caption>There are no additional arguments passed for this function call.</Caption>
+      )}
+    </>
+  );
+}
+
+export default memo(FunctionArgumentsList);
```

Comment:
I did realise after suggesting to use `function` defined component, there's a slightly harder case when needing to wrap/memoize components.

We should avoid default exports, though. So maybe we can do something like this.

```suggestion
export const FunctionArgumentsList = memo(FunctionArgumentsListBase);
```

Reply:
O, great, thx for the suggestion! I didn't think to do it that way for some reason and knew what I did wasn't quite what we wanted. 👍